packages/http/src/client.ts

```diff
@@ -0,0 +1,141 @@
+import { IPrismOutput } from '@stoplight/prism-core';
+import { IHttpOperation } from '@stoplight/types';
+// @ts-ignore
+import logger from 'abstract-logging';
+import { getOrElse } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { getStatusText } from 'http-status-codes';
+import { defaults } from 'lodash';
+import { parse as parseQueryString } from 'querystring';
+import { parse as parseUrl } from 'url';
+import { createInstance } from '.';
+import { forwarder } from './forwarder';
+import getHttpOperations, { getHttpOperationsFromResource } from './getHttpOperations';
+import { mocker } from './mocker';
+import { router } from './router';
+import { IHttpConfig, IHttpRequest, IHttpResponse, IHttpUrl } from './types';
+import { validator } from './validator';
+
+async function createClientFromResource(specFile: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
+  const resources = await getHttpOperationsFromResource(specFile);
+  return createClientFromOperations(resources, defaultConfig);
+}
+
+async function createClientFromString(spec: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
+  const resources = await getHttpOperations(spec);
+  return createClientFromOperations(resources, defaultConfig);
+}
+
+function createClientFromOperations(resources: IHttpOperation[], defaultConfig: IHttpConfig) {
+  const obj = createInstance(defaultConfig, {
+    logger,
+    router,
+    forwarder,
+    validator,
+    mocker,
+  });
+
+  const request: RequestFunction = async (url, input, config) => {
+    const parsedUrl = parseUrl(url);
+
+    if (!parsedUrl.pathname) throw new Error('path name must alwasy be specified');
+
+    const httpUrl: IHttpUrl = {
+      baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : undefined,
+      path: parsedUrl.pathname,
+      query: parseQueryString(parsedUrl.query || ''),
+    };
+
+    const data = await obj.request(
+      {
+        ...input,
+        url: httpUrl,
+      },
+      resources,
+      config,
+    );
+
+    const output: PrismOutput = {
+      status: data.output.statusCode,
+      statusText: pipe(
+        data.output.statusText,
+        getOrElse(() => getStatusText(data.output.statusCode)),
+      ),
+      headers: data.output.headers || {},
+      data: data.output.body || {},
+      config: defaults(config, defaultConfig),
+      request: { ...input, url: httpUrl },
+      validations: data.validations,
+    };
+
+    return output;
+  };
+
+  function isInput(input?: Pick<IHttpRequest, 'headers'> | IHttpConfig): input is Pick<IHttpRequest, 'headers'> {
```

Comment:
if there's no reason to keep it inside `createClientFromOperations`, then let's move it out

Reply:
Answered in the comment above.